src/emsarray/conventions/_base.py

```diff
@@ -1286,6 +1286,24 @@ def polygons(self) -> numpy.ndarray:
         :meth:`ravel_index`
         :attr:`mask`
         """
+        polygons = self._make_polygons()
+
+        not_none = (polygons != None)  # noqa: E711
+        invalid_polygon_indices = numpy.flatnonzero(not_none & ~shapely.is_valid(polygons))
```

Comment:
TODO: For SHOC datasets with large swathes of 'missing' polygons, check that `shapely.is_valid()` is performant at skipping these holes.

Reply:
Doing an `is_valid()` check on an empty array is quite performant:

```python
In [1]: import numpy
   ...: import shapely
   ...: 
   ...: xx, yy = numpy.meshgrid(numpy.arange(1001), numpy.arange(1001))
   ...: grid = numpy.stack([xx, yy], axis=-1)
   ...: points = numpy.stack([
   ...:     grid[:-1, :-1],
   ...:     grid[:-1, +1:],
   ...:     grid[+1:, +1:],
   ...:     grid[+1:, :-1],
   ...: ], axis=2).reshape((-1, 4, 2))
   ...: polygons = shapely.polygons(points)
   ...: 
   ...: %timeit shapely.is_valid(polygons)
776 ms ± 177 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [2]: polygons = numpy.full(1000000, dtype=object, fill_value=None)
   ...: polygons[987654] = shapely.geometry.box(0, 0, 1, 1)
   ...: 
   ...: %timeit shapely.is_valid(a)
2.69 ms ± 30.4 μs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```